mlflow/tracking/fluent.py

```diff
@@ -3053,32 +3053,67 @@ def setup_autologging(module):
             register_post_import_hook(setup_autologging, "pyspark.ml", overwrite=True)
 
 
-_ACTIVE_MODEL_ID = ContextVar("active_model_id", default=None)
+class ActiveModelContext:
+    """
+    The context of the active model.
+
+    Args:
+        model_id: The ID of the active model.
+        set_by_user: Whether the active model was set by the user or not.
+    """
+
+    def __init__(self, model_id: Optional[str] = None, set_by_user: bool = False):
+        self.model_id = model_id
+        self.set_by_user = set_by_user
```

Comment:
```suggestion
        self._model_id = model_id
        self._set_by_user = set_by_user
```

can we set `_model_id` and `_set_by_user` instead?

Reply:
This invokes the property setter so it should be the same?

Reply:
ah you're right. Never mind this comment.